src/routes/blog/post/chatbot-with-webllm-and-webgpu/+page.markdoc

```diff
@@ -0,0 +1,474 @@
+}
+
+h1 {
+  font-size: 1.5rem;
+  font-weight: 600;
+  margin: 0 0 1.5rem;
+  color: #111827;
+}
+
+#output {
+  background: #f3f4f6;
+  padding: 1.25rem;
+  min-height: 220px;
+  border-radius: 12px;
+  white-space: pre-wrap;
+  overflow-y: auto;
+  max-height: 420px;
+  font-size: 0.95rem;
+}
+
+input,
+button,
+select {
+  font-family: inherit;
+  font-size: 0.95rem;
+  padding: 0.8rem 1rem;
+  width: 100%;
+  border-radius: 10px;
+  border: 1px solid #e5e7eb;
+  background-color: white;
+}
+
+button {
+  background-color: #fd366e;
+  color: white;
+  border: none;
+  font-weight: 500;
+  cursor: pointer;
+  transition: background-color 0.15s;
+}
+
+button:hover:not(:disabled) {
+  background-color: #e62e60;
+}
+
+button:disabled {
+  background-color: #ffa5c0;
+  cursor: not-allowed;
+}
+
+.error {
+  color: #dc2626;
+  background-color: #fee2e2;
+  padding: 0.8rem;
+  border-radius: 10px;
+  margin-top: 0.75rem;
+  font-size: 0.9rem;
+}
+
+.controls {
+  display: grid;
+  grid-template-columns: 2fr 1fr;
+  gap: 0.75rem;
+  margin-bottom: 1.5rem;
+}
+
+.chat-container {
+  margin-top: 1.5rem;
+  display: flex;
+  flex-direction: column;
+  gap: 1.5rem;
+}
+
+.progress-container {
+  margin-top: 1rem;
+}
+
+.progress-bar {
+  width: 100%;
+  height: 8px;
+  background-color: #e5e7eb;
+  border-radius: 999px;
+  overflow: hidden;
+}
+
+.progress-fill {
+  height: 100%;
+  background-color: #fd366e;
+  width: 0%;
+  transition: width 0.3s ease;
+}
+
+.progress-text {
+  font-size: 0.8rem;
+  text-align: center;
+  margin-top: 0.5rem;
+  color: #6b7280;
+}
+
+.form-group {
+  display: flex;
+  flex-direction: column;
+  gap: 0.75rem;
+}
+
+input {
+  box-sizing: border-box;
+  margin: 0rem;
+}
+
+input:focus,
+select:focus {
+  outline: none;
+  border-color: rgba(253, 54, 110, 0.5);
+  box-shadow: 0 0 0 1px rgba(253, 54, 110, 0.1);
+}
+
+```
+
+Next, open the `index.html` file in your browser and you should see something like this:
+
+![WebLLM and WebGPU chat app demo](/images/blog/chatbot-with-webllm-and-webgpu/webllm-webgpu-chat-app-demo.gif)
+
+Try loading a model and chatting with the AI. To be sure that this all runs locally, you can also turn off your internet connection and see if the app still works after initial model download.
+
+## Conclusion
+
+We've built a local chatbot that runs entirely in the browser. You can now load a model, chat with it, and have a real-time conversation, all without a server.
+
+This is the future of AI. Local, private, and fast. And what we've done is just the beginning. You can build on this foundation, add chat history, improve UX, support longer contexts, or experiment with your own compiled models.
+
+You can also check out a more complex version of this app that includes chat history and a few other features [here](https://github.com/ebenezerdon/webllm-offline-ai).
+
+The repo for what we built in this guide is available at https://github.com/appwrite-community/simple-webllm-chat.
```

Comment:
```suggestion
The source code for the app we built in this guide is available on [GitHub](https://github.com/appwrite-community/simple-webllm-chat).
```

Just reframing plus it lets the link be clickable

Reply:
That's better. Thanks